lib/spack/spack/installer.py

```diff
@@ -374,6 +374,13 @@ def _update_explicit_entry_in_db(pkg, rec, explicit):
             rec.explicit = True
 
 
+def clear_failures():
+    """
+    Remove all failure tracking markers for the Spack instance.
+    """
+    spack.store.db.clear_all_failures()
```

Comment:
not opposed to this function but it does seem to indicate that the failure file management should get its own class.

Reply:
True.  Where does the separate class fit within the priorities?